addons/mail/static/tests/qunit_suite_tests/components/attachment_list_tests.js

```diff
@@ -146,6 +146,50 @@ QUnit.test('simplest layout + editable', async function (assert) {
     );
 });
 
+QUnit.test('link-type attachment should have open button instead of download button', async function (assert) {
+    assert.expect(3);
+
+    const pyEnv = await startServer();
+    const channelId = pyEnv['mail.channel'].create({
+        channel_type: 'channel',
+        name: 'channel1',
+    });
+    const messageAttachmentId = pyEnv['ir.attachment'].create({
+        name: "url.example",
+        mimetype: 'text/plain',
+        type: 'url',
+        url: 'https://www.odoo.com',
+    });
+    pyEnv['mail.message'].create({
+        attachment_ids: [messageAttachmentId],
+        body: "<p>Test</p>",
+        model: 'mail.channel',
+        res_id: channelId
+    });
+    const { openDiscuss } = await start({
+        discuss: {
+            context: { active_id: channelId },
+        },
+    });
+    await openDiscuss();
+
+    assert.strictEqual(
+        document.querySelectorAll(`.o_AttachmentCard_asideItemDownload`).length,
+        0,
+        "attachment should not have a download button"
+    );
+    assert.strictEqual(
+        document.querySelectorAll(`.o_AttachmentCard_asideItemOpenLink`).length,
+        1,
+        "attachment should have a open link button"
+    );
+    assert.strictEqual(
+        document.querySelector(`.o_AttachmentCard_asideItemOpenLink`).target,
+        '_blank',
+        "attachment should have a open link button in a new tab"
+    );
```

Comment:
Rendering is not necessarily finished after `await openDiscuss()`. Prefer using `contains()`